src/System.CommandLine/Binding/BindingContext.cs

```diff
@@ -63,6 +63,15 @@ public ModelBinder GetModelBinder(IValueDescriptor valueDescriptor)
             return new ModelBinder(valueDescriptor);
         }
 
+        internal ModelBinder GetModelBinder(Type type)
+        {
+            if (_modelBindersByValueDescriptor.TryGetValue(type, out ModelBinder binder))
+            {
+                return binder;
+            }
+            return new ModelBinder(type);
```

Comment:
Is it deliberate that we don't cache this?

Reply:
Method removed as dead code